### PR TITLE
Possible code readability improvement for Jetpack-related view models

### PR DESCRIPTION
**Pull request by hafizrahman**
### Description

Under the hood, Jetpack activation during REST API login is quite a complex process. Last week while doing groundskeeping on the project, I figured I could improve the code readability in the way Jetpack status is handled prior to activation process.

The entire flow is the same, I'm mostly just taking out some repetitive parts and putting it into its own function, as well as reducing some verbosity (e.g: by destructuring `Pair`).

I don't think this is a *must*, but I'm curious if y'all think this is worth adding it in?

### CODE REVIEW

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/jetpack/benefits/JetpackActivationEligibilityErrorViewModel.kt [hunk 1]
```diff
@@ -64,32 +64,36 @@ class JetpackActivationEligibilityErrorViewModel @Inject constructor(
...
-                                }
-                            },
-                            onFailure = { /* Still not eligible, keep showing error screen */ }
-                        )
-                    }
+            onSuccess = { (jetpackStatus, fetchResponse) ->
+                when (fetchResponse) {
+                    JetpackStatusFetchResponse.SUCCESS -> handleSuccess(jetpackStatus)
+                    JetpackStatusFetchResponse.NOT_FOUND -> checkUserEligibility(jetpackStatus)
+                    else -> { /* Still not eligible, keep showing error screen */ }
```
Comment (JorgeMucientes) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/jetpack/benefits/JetpackActivationEligibilityErrorViewModel.kt:96]: Nice! I like it!

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/jetpack/benefits/JetpackBenefitsViewModel.kt [hunk 2]
```diff
@@ -74,78 +79,58 @@ class JetpackBenefitsViewModel @Inject constructor(
...
+                )
+            }
+        }
+
+        result.fold(
+            onSuccess = { (jetpackStatus, fetchResponse) ->
+                when (fetchResponse) {
+                    JetpackStatusFetchResponse.SUCCESS -> {
+                        startJetpackActivation(jetpackStatus)
+                        logSuccess(Pair(jetpackStatus, fetchResponse))
```
Comment (JorgeMucientes) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/jetpack/benefits/JetpackBenefitsViewModel.kt]: Np, maybe you could refactor this function as well to get rid of the unnecessary pair. Something like:
```kotlin
    private fun logSuccess(jetpackStatus: JetpackStatus) {
        if (isAppPasswords) {
            analyticsTrackerWrapper.track(
                stat = JETPACK_SETUP_CONNECTION_CHECK_COMPLETED,
                properties = mapOf(
                    AnalyticsTracker.KEY_JETPACK_SETUP_IS_ALREADY_CONNECTED to jetpackStatus.isJetpackConnected,
                    AnalyticsTracker.KEY_JETPACK_SETUP_REQUIRES_CONNECTION_ONLY to
                        (jetpackStatus.isJetpackInstalled && !jetpackStatus.isJetpackConnected)
                )
            )
        }
    }
```

Which by the way could also be a nested function of `handleJetpackStatusResult`. But tbh I don't have a major preference with that.

**Review by JorgeMucientes — APPROVED** [review]

**Comment by JorgeMucientes** [comment]
Nice refactor @hafizrahman. I think the code is definitely more readable with this changes. Thanks for working on that. I left a minor suggestion you might want to apply. If not :shipit:

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/jetpack/benefits/JetpackBenefitsViewModel.kt (hunk 2 above)
Comment (hafizrahman) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/jetpack/benefits/JetpackBenefitsViewModel.kt]: Great eye, thanks! I'll add it.

Also TIL that Android Studio can't tell if something in a Pair function parameter is unused.

Comment (JorgeMucientes) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/jetpack/benefits/JetpackBenefitsViewModel.kt]: What actually caught my attention was the use of `Pair` for passing function params. I understand the reason to use `Pair` to return couple values from a function that only allows you to return a single result. But I'm curious what was the reason to have a `Pair` for function parameters that you can easily pass individually without adding extra nesting structures?

**Pull request merged**